commonui/src/main/java/com/hyperwallet/android/ui/common/insight/HyperwalletInsight.java

```diff
@@ -18,34 +18,229 @@
 package com.hyperwallet.android.ui.common.insight;
 
 import android.content.Context;
+import android.os.Handler;
 
 import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.VisibleForTesting;
+
+import com.hyperwallet.android.Configuration;
+import com.hyperwallet.android.Hyperwallet;
+import com.hyperwallet.android.HyperwalletAuthenticationTokenProvider;
+import com.hyperwallet.android.exception.HyperwalletException;
+import com.hyperwallet.android.insight.Insight;
+import com.hyperwallet.android.insight.collect.ErrorInfo;
+import com.hyperwallet.android.listener.HyperwalletListener;
 
 import java.util.Map;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
 
 /**
  * Used for gathering the data necessary for the Insights analytics.
  */
 public class HyperwalletInsight {
```

Comment:
**Suggestion:** 
User can break HyperwalletInsight singleton class based on following technique,
1)Reflection
2)Serialization
Reference link https://www.geeksforgeeks.org/prevent-singleton-pattern-reflection-serialization-cloning/

Reply:
I don't think we need to worry about this, as achieving this is fairly hacky. If the integrator wants to break their own application it's up to them 🤷‍♂